ddp-workspace/projects/ddp-dsm-ui/src/app/FON/components/register-patients-modal/register-patients-modal.component.ts

```diff
@@ -1,60 +1,113 @@
 
 @Component({
   selector: 'app-add-patients-modal',
   templateUrl: './register-patients-modal.component.html',
   styleUrls: ['./register-patients-modal.component.scss'],
 })
-export class RegisterPatientsModalComponent implements OnInit {
-  uploading = false;
+export class RegisterPatientsModalComponent implements OnInit, OnDestroy {
+  isAddingPatient: boolean;
+  addedPatients: number;
 
-  readonly addPatientForm: FormGroup = this.formBuilder.group({
+  readonly patientsHttpArray: Observable<any>[] = [];
+  readonly patientsAddingForm: FormGroup = this.formBuilder.group({
     patients: this.formBuilder.array([])
   });
 
-  constructor(private dialogRef: MatDialogRef<RegisterPatientsModalComponent>,
-              private formBuilder: FormBuilder) {
+  unsubSubject = new Subject<void>();
+
+  @ViewChild('successMessage') successMessageTemplate: TemplateRef<any>;
+
+  constructor(
+    private dialogRef: MatDialogRef<RegisterPatientsModalComponent>,
+    private formBuilder: FormBuilder,
+    private patientsService: PatientsService,
+    private httpService: HttpService,
+    private _snackBar: MatSnackBar
+  ) {
   }
 
   ngOnInit(): void {
     this.addPatient();
   }
 
+  ngOnDestroy(): void {
+    this.unsubSubject.next();
+  }
+
   public get patients(): FormArray {
-    return this.addPatientForm.controls.patients as FormArray;
+    return this.patientsAddingForm.controls.patients as FormArray;
   }
 
   public addPatient(): void {
-    const patient = this.formBuilder.group({
-      email: new FormControl(null, [Validators.required, Validators.email]),
-      firstName: new FormControl(null, Validators.required),
-      lastName: new FormControl(null, Validators.required),
-      dateOfBirth: new FormControl(null, Validators.required),
-      informedConsentDate: new FormControl(null, Validators.required),
-      assentDate: new FormControl(null)
-    });
-
-    this.patients.push(patient);
+    this.patients.push(this.patientFormGroup);
   }
 
   public removePatient(index: number): void {
     this.patients.removeAt(index);
   }
 
   public registerPatient(): void {
-    console.log(this.addPatientForm.getRawValue());
-    this.uploading = true;
-    setTimeout(() => {
-      this.closeDialog();
-      this.uploading = false;
-    }, 2000);
-
+    this.isAddingPatient = true;
+    this.patientsAddingForm.getRawValue()
+      .patients
+      .forEach(patient => {
+          const generatedPatientInfo = this.patientsService.generatePatientInfo(patient);
+          this.patientsHttpArray.push(this.getPatientHttp(generatedPatientInfo));
+        }
+      );
+    this.uploadPatients();
   }
 
   public closeDialog(): void {
     this.dialogRef.close();
   }
+
+  private uploadPatients(): void {
+    forkJoin(this.patientsHttpArray)
+      .pipe(takeUntil(this.unsubSubject))
+      .subscribe((resultPatients) => {
+        this.addedPatients = resultPatients
+          .filter(patient => !(patient instanceof HttpErrorResponse))
```

Comment:
I hope it will be replaced with more proper error handler when we have an API for adding multiple patients at one time.

Reply:
Of course, I even thought about a global error handler, which will pop up like the success message does now, but with red in red colors, saying what error has happened.